spec/services/users/delete_account_service_spec.rb

```diff
@@ -0,0 +1,91 @@
+  let!(:coach_team_enrollment) { create :faculty_startup_enrollment, startup: team_2, faculty: coach }
+  let!(:coach_note_by_user) { create :coach_note, author: user, student: team_2.founders.first }
+  let!(:connect_slot) { create :connect_slot, faculty: coach }
+  let!(:submission_reviewed_by_user) { create :timeline_event, :with_owners, latest: true, owners: team_2.founders, target: target_1, evaluator: coach, evaluated_at: Time.zone.now }
+
+  # Create community records
+  let!(:post) { create :post, creator: user, topic: create(:topic) }
+  let!(:post_like) { create :post_like, user: user, post: post }
+
+
+  # Miscellaneous records
+  let!(:issued_certificate_for_user) { create :issued_certificate, user: user }
+  let!(:markdown_attachment_by_user) { create :markdown_attachment, user: user }
+  let!(:course_export) { create :course_export, :teams, user: user, course: course }
+
+  describe '#execute' do
+    context 'user is an admin in school' do
+      before do
+        create(:school_admin, user: user, school: user.school)
+      end
+      it 'an exception is raised' do
+        expect { subject.execute }.to raise_error('user is a school admin')
+      end
+    end
+
+    context 'user is not an admin in school' do
+      before do
+        post.text_versions.create!(value: post.body, user: user, edited_at: post.updated_at)
+      end
+      it 'deletes data and nullifies references in applicable records' do
+        subject.execute
+
+        # Check only applicable records are destroyed
+        expect(User.find_by(id: user.id)).to eq(nil)
+        expect(Founder.find_by(id: student.id)).to eq(nil)
+        expect(Startup.find_by(id: student_team.id)).to eq(nil)
+        expect(Startup.find_by(id: team_with_user.id)).to_not eq(nil)
+        expect(TimelineEventOwner.where(founder_id: user.founders.select(:id))).to eq([])
+        expect(TimelineEvent.find_by(id: submission_by_student.id)).to eq(nil)
+        expect(TimelineEvent.find_by(id: submission_by_team.id)).to_not eq(nil)
+        expect(submission_by_team.founders.pluck(:id).sort).to eq(team_with_user.founders.reload.pluck(:id).sort)
+        expect(CourseAuthor.find_by(id: course_author.id)).to eq(nil)
+        expect(Faculty.find_by(id: coach.id)).to eq(nil)
+        expect(CoachNote.find_by(id: coach_note_1.id)).to eq(nil)
+        expect(FacultyCourseEnrollment.find_by(id: coach_course_enrollment.id)).to eq(nil)
+        expect(FacultyStartupEnrollment.find_by(id: coach_team_enrollment.id)).to eq(nil)
+        expect(ConnectSlot.find_by(id: connect_slot.id)).to eq(nil)
+
+        # Check user_id is nullified in applicable records
+        expect(coach_note_by_user.reload.author_id).to eq(nil)
+        expect(submission_reviewed_by_user.reload.evaluator_id).to eq(nil)
+        expect(post.reload.creator_id).to eq(nil)
+        expect(post_like.reload.user_id).to eq(nil)
+        expect(post.text_versions.reload.first.user_id).to eq(nil)
```

Comment:
This `reload` shouldn't be required, since it's on a relation, which will always cause a query.